public/app/features/dashboard/components/SaveDashboard/SaveDashboardDrawer.tsx

```diff
@@ -27,6 +28,11 @@ export const SaveDashboardDrawer = ({ dashboard, onDismiss, onSaveSuccess, isCop
       return undefined;
     }
 
+    // If dashboardJson is set, it means that we're in the dashboard embed view
+    if (dashboardJson) {
+      return dashboardJson;
+    }
+
     const result = await backendSrv.getDashboardByUid(dashboard.uid);
```

Comment:
I don't think we need the getJson() we should just be able to have an original object somewhere (that should be the same no matter the loading source). 

The diff should not look at the server version like it does today, that creates really strange situations where you there is no diff even when you have made a change (say if someone made same change and saved it)

Reply:
I tried saving json into the reducer without function but that freezes the object, which breaks panel logic for some reason.

Reply:
What about saving the original dashboard in `dashboardInitCompleted`? Then it can be retrieved from the Redux state when needed.

Reply:
@Clarity-89 

> I tried saving json into the reducer without function but that freezes the object, which breaks panel logic for some reason, apparently it tries modifying it directly.

ah, I think we remove all the properties we don't want to count as a change (time range, variable values etc) so they don't show up in the diff